test/functional/mochitest/browser_as_private_browsing.js

```diff
@@ -0,0 +1,30 @@
+"use strict";
+
+let Cu = Components.utils;
+Cu.import("resource://gre/modules/Task.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+/**
+ * Tests that loading activity streams in private browsing mode doesn't actually
+ * load the activity streams URL and instead loads about:privatebrowsing
+ */
+add_task(function*doesnt_load_in_private_windows() {
+  const asURL = "resource://activity-streams/data/content/activity-streams.html#/";
+  const privateBrowsingURL = "about:privatebrowsing";
+
+  // create a private browsing window
+  let privateWindow = yield BrowserTestUtils.openNewBrowserWindow({private: true});
```

Comment:
https://dxr.mozilla.org/mozilla-beta/source/testing/mochitest/BrowserTestUtils/BrowserTestUtils.jsm#471 makes me wonder if it's worth running this test both with and without remote set to true so that we guarantee correctness both with and without e10s.